tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-navigation-access.bal

```diff
@@ -128,3 +128,23 @@ function testXMLNavigationExpressionWithQuotedIdentifiers() returns [xml, xml] {
 
     return [x/<'object>, x/<ns0:'object>];
 }
+
+function testXMLNavigationExpressionWithXMLSubtypeOnLHS() {
+        xml val = xml `<foo><bar>0</bar><baz>1</baz><bar>2</bar></foo>`;
+        xml<xml:Element> xmlResult = val/<bar>;
+        string s = xmlResult.toString();
+        if (s == "<bar>0</bar><bar>2</bar>") {
```

Comment:
Any particular reason to assert the string representation instead of checking equality between two sequences? Same below.